websites/N/Nebula/presence.ts

```diff
@@ -0,0 +1,261 @@
+		presenceData.buttons = [
+			{
+				label: "Watch Video",
+				url: document.location.href,
+			},
+			{
+				label: "View channel",
+				url:
+					getRootUrl() +
+					document.querySelector(".css-1nacd6b").getAttribute("href"),
+			},
+		];
+	}
+}
+
+function getSearchDetails(presenceData: PresenceData): void {
+	presenceData.details = "Searching for:";
+	presenceData.state = parseQueryParams().q || "...";
+}
+
+function getOtherDetails(
+	presenceData: PresenceData,
+	showButtons: boolean
+): void {
+	const videoElement = document.querySelector("video"),
+		audioElement = document.querySelector("audio");
+
+	if (videoElement === null && audioElement === null) {
+		// viewing a channel
+		const channelName = document.querySelector(".css-dv828b"),
+			podcastName = document.querySelector(".css-xdl2kn");
+
+		if (channelName === null && podcastName === null) return;
+
+		if (channelName === null) {
+			presenceData.details = "Viewing a podcast";
+			presenceData.state = podcastName.innerHTML;
+		} else {
+			presenceData.details = "Viewing a channel";
+			presenceData.state = channelName.innerHTML;
+		}
+
+		if (showButtons) {
+			presenceData.buttons = [
+				{
+					label: "View channel",
+					url: document.location.href,
+				},
+			];
+		}
+	} else if (videoElement === null) {
+		//its a podcast
+		const channelElement = document.querySelector(".css-2m8aus");
+
+		presenceData.details = document.querySelector(".css-l7qxoj").innerHTML;
+		presenceData.state = channelElement.innerHTML;
+		setTimestamps(audioElement, presenceData);
+
+		if (showButtons) {
+			presenceData.buttons = [
+				{
+					label: "Listen to podcast",
+					url: document.location.href,
+				},
+				{
+					label: "View channel",
+					url: getRootUrl() + channelElement.getAttribute("href"),
+				},
+			];
+		}
+	} else {
+		//its a class episode
+		const episodeName = document.querySelector(".css-13igzay"),
+			className = document.querySelector(".css-1ls4t7r");
+
+		if (episodeName !== null)
+			presenceData.details = `${episodeName.innerHTML} | ${className.innerHTML}`;
+		else presenceData.details = className.innerHTML;
+
+		presenceData.state = document.querySelector(".css-p7br9k").innerHTML;
+		setTimestamps(videoElement, presenceData);
+
+		if (showButtons) {
+			presenceData.buttons = [
+				{
+					label: "Watch Episode",
+					url: document.location.href,
+				},
+			];
+		}
+	}
+}
+
+function setTimestamps(
+	element: HTMLAudioElement | HTMLVideoElement,
+	presenceData: PresenceData
+): void {
+	delete presenceData.startTimestamp;
+	presenceData.endTimestamp = presence.getTimestampsfromMedia(element)[1];
+	if (element.paused) {
+		delete presenceData.endTimestamp;
+		presenceData.smallImageKey = Assets.Pause;
+	} else presenceData.smallImageKey = Assets.Play;
+}
+
+interface QueryParams {
+	[key: string]: string;
+}
+
+function parseQueryParams(): QueryParams {
+	const queryParams: QueryParams = {},
+		queryString = window.location.search.split("?")[1];
```

Comment:
Use `document.location` for consistency
```suggestion
		queryString = document.location.search.split("?")[1];
```